workshops/reddit_viewer/README.md

```diff
@@ -0,0 +1,126 @@
+
+In your repl.it open your .env file and make a variable called `SECRET` and another one called `ID`. Now go back to your reddit apps section and copy the secret from the secret part to your SECRET variable in your env file. Also copy the section just under your personal use script to your ID variable.
+
+![](https://cloud-119djxpws.vercel.app/image.png)
+
+We have the authentication setup which is easier done than said. Now one last step.
+
+We need to make a reddit variable that connects to the reddit api. Make a reddit = praw.Reddit() function. Now the next steps all go into the parenthesis. All it requires is three things, `client_id` which is your `"ID"` variable. So, do `client_id=os.getenv("ID"),`. The next one is client_secret. `client_secret=os.getenv("SECRET"),`. The final one is user_agent which is just a fancy name for what browser on what device. We will use a sample one as it doesn't actually matter which one you use. `user_agent="Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0")`
+
+![What the connection to Reddit should look like](https://cloud-ggp6iugz2.vercel.app/image.png)
+
+To test it out, run `print(reddit.read_only)`, if it returns true then you are golden.
+
+# Step 3: Setting up Flask Pages
+
+Make a folder called `templates` and inside of `templates` add an `index.html` file.
+
+![Your folder structure for the rest of this workshop](https://cloud-qvylewnf8.vercel.app/image.png)  
+Setup this index.html like any other html file with a header and a body.
+
+We want to load this index.html into our view. So, make an `<h1>` in the body that just says, `"Hello World"`. Now go to the main.py file and add an `@app.route('/')`. This means that the root (/) should load whatever function is under it. Now directly underneath the @app.route, add a function called front_page that returns render_template('index.html').
+
+![The front_page method](https://cloud-6x6i6q8oh.vercel.app/image.png).
+
+Now press run and you should see a Hello World.
+
+# Step 4: Getting and handling user input
+
+Go to the index.html file and add a `form` with a `method="POST"`. In the form area add an `<h3>` that just has something like `Go to subreddit`. In the same form area add an input with the `name="subreddit"`. Also have the `input type=submit`. Thats it from the html side of user input. Go to the main.py file and add another @app.route going to the ('/') but this time add a comma and have a methods=["POST."] `@app.route('/', methods=['POST'])`. This means that when the html sends a POST event, we will handle it in the function. Under the app.route, add a new method called `front_page_post`. In that function remember to put a `return render_template('index.html')` put keep this at the bottom of the function always.
```

Comment:
Same as before, make this step a lot more detailed. Remove the images that are just pictures of code and replace with them with code blocks that build up on each other and clearly demonstrate each step. Otherwise the instructions are very confusing

Reply:
Done!